### PR TITLE
Object Detection Model Wrapper Support 

### PR DESCRIPTION
**Pull request by natalie-isak**
This PR creates support for the object detection scenario. With this new ML model wrapper, users can easily call predict and predict_proba on object detection ML models.

### CODE REVIEW

File: python/ml_wrappers/model/image_model_wrapper.py [hunk 1]
```diff
@@ -14,6 +14,9 @@
 from ml_wrappers.model.pytorch_wrapper import WrappedPytorchModel
 from ml_wrappers.model.wrapped_classification_model import \
     WrappedClassificationModel
+import torch
+import torchvision
```
Comment (Advitya17) [inline python/ml_wrappers/model/image_model_wrapper.py:18]: Curious if `torchvision` is in the dependencies and/or needs to be added?

File: python/ml_wrappers/model/image_model_wrapper.py [hunk 2]
```diff
@@ -232,3 +237,52 @@ def predict_proba(self, dataset: pd.DataFrame) -> np.ndarray:
...
         return np.stack(predictions.probs.values)
+    
+class WrappedObjectDetectionModel:
+    """A class for wrapping a object detection model."""
+
+    def __init__(self, model) -> None:
+        """Initialize the WrappedObjectDetectionModel with the model and evaluation function.
+
+        :param model: mlflow model
+        :type model:
```
Comment (Advitya17) [inline python/ml_wrappers/model/image_model_wrapper.py:248]:
```suggestion
        :type model: Any
```

Comment (Advitya17) [inline python/ml_wrappers/model/image_model_wrapper.py:248]:
```suggestion
        :type model: Any
```

File: python/ml_wrappers/model/image_model_wrapper.py (hunk 1 above)
Comment (natalie-isak) [reply python/ml_wrappers/model/image_model_wrapper.py]: it is not in the dependency file. @imatiach-msft would you recommend adding to dependencies?

**Comment by imatiach-msft** [comment]
Can you please try to rebase to latest?  The macos tests were disabled in the repo, they were failing.  I am trying to fix them in PR https://github.com/microsoft/ml-wrappers/pull/89 but the builds should be passing now.  Also, it looks like the linting build is still failing, you need to fix the linting issues.

File: python/ml_wrappers/model/image_model_wrapper.py [hunk 3]
```diff
@@ -232,3 +240,61 @@ def predict_proba(self, dataset: pd.DataFrame) -> np.ndarray:
         """
         predictions = self._mlflow_predict(dataset)
         return np.stack(predictions.probs.values)
+
+
+class WrappedObjectDetectionModel:
```
Comment (imatiach-msft) [inline python/ml_wrappers/model/image_model_wrapper.py:245]: this looks great!  can you please add tests to the new wrapper and perhaps even add some documentation on the predict function specification for object detection?  maybe the docs can be added in a separate PR and we can strategize on where to put them in the repo.  I imagine we would want to put them in some place like:
https://github.com/microsoft/ml-wrappers/tree/main/docs

File: python/ml_wrappers/model/image_model_wrapper.py [hunk 4]
```diff
@@ -22,14 +26,13 @@
 try:
     import torch.nn as nn
 except ImportError:
-    module_logger.debug('Could not import torch, required if using a PyTorch model')
+    module_logger.debug("Could not import torch, required if using a PyTorch model")
```
Comment (imatiach-msft) [inline python/ml_wrappers/model/image_model_wrapper.py:29]: Was this auto-changed somehow?  I think we try to stick to single quotes everywhere across all RAI repos for strings in python code.

**Review by imatiach-msft — APPROVED** [review]
Approved!  Looks great!  Please add tests before merging.

File: python/ml_wrappers/model/image_model_wrapper.py [hunk 5]
```diff
@@ -8,12 +8,16 @@
 
 import numpy as np
 import pandas as pd
+import torch
```
Comment (imatiach-msft) [inline python/ml_wrappers/model/image_model_wrapper.py:11]: please put these below in the try/catch.  pytorch is NOT required for this package, since we want this package to be used by users even if they don't have pytorch installed.  If you don't move this to the try/catch, users without pytorch will see import of ml_wrappers package error out since they don't have pytorch.

File: python/ml_wrappers/model/image_model_wrapper.py (hunk 1 above)
Comment (imatiach-msft) [reply python/ml_wrappers/model/image_model_wrapper.py]: Actually, I forgot, we shouldn't depend on it in ml-wrappers anyway since users should be able to use this package without pytorch.  So we definitely shouldn't have it in the setup.py.  We also wrap all imports in the ml-wrappers package from other frameworks like pytorch, tensorflow, xgboost etc in try/catch since the imports should not cause ml_wrappers package as a whole to fail.

Comment (natalie-isak) [reply python/ml_wrappers/model/image_model_wrapper.py]: Got it - so to summarize, torch/torchvision etc should not be added to requirements.txt, but can still be imported in individual files w/in ml-wrappers

File: python/ml_wrappers/model/image_model_wrapper.py (hunk 3 above)
Comment (natalie-isak) [reply python/ml_wrappers/model/image_model_wrapper.py:245]: Working on the tests now. As for documentation, would I add it here? https://github.com/microsoft/ml-wrappers/blob/main/docs/WrapperSpecifications.md Thanks!

Comment (imatiach-msft) [reply python/ml_wrappers/model/image_model_wrapper.py:249]: yes we can add automl support + tests in a separate PR, I am guessing more changes might be needed to support their models?

**Review by Advitya17 — CHANGES_REQUESTED** [review]
May need to make changes to propagate labels.

**Review by Advitya17 — APPROVED** [review]

**Review by Advitya17 — APPROVED** [review]

**Review by Advitya17 — APPROVED** [review]

**Review by Advitya17 — APPROVED** [review]

**Review by Advitya17 — APPROVED** [review]
